### ISSUE TITLE
Weights from TF-based CARE to CAREAMICS

### ISSUE THREAD

**Issue by OReyesMatte** [CAREamics/careamics#263]
I have trained some models using the original CARE framework for image deconvolution and fluorescence translation. Is it possible to take the parameters of those models and implement them in CAREAMICS? Or should I train a new model using the same data?

Thanks in advance!

**Comment by jdeschamps** [comment]
Hi!

If you are referring to the weights of the model, then right now there is no way to do that.  You would need to retrain from scratch.

One reason is the model themselves are different (not exactly the same architecture). The other is that we would need to implement a conversion and that is not really planned at the moment.

We are currently examining whether we should use strictly the same architecture (I just opened that issue https://github.com/CAREamics/careamics/issues/264 to make it obvious we have this in mind).

If you can retrain your models and compare the performances, we would be super interested in knowing how well CAREamics does in comparison with CSBDeep on your data. That might also push us to prioritize adding the exact CSBDeep architecture in CAREamics.

Don't hesitate to open issues if you encounter reproducibility issues between CSBdeep and CAREamics, we have been mostly focused on N2V but we are currently implementing CARE examples.

**Comment by OReyesMatte** [comment]
Thanks a lot! I will run some tests and then see how it performs

**Comment by OReyesMatte** [comment]
I ran a test for a CARE-based deconvolution model I trained, and I have to say, that the results from CAREAMICS are nearly identical! Even when using **much less** training data, the predicted deconvolutions are basically the same. To further test it, I compared the segmentation of a custom model for segmenting deconvolved membranes, and visually there appears to be no difference between the two images. This is fantastic as I went from using 200 images of training data to only _10_, and while maybe it was too much data at the beginning, the results are nonetheless beautiful with only a few training images. I attach example covers of both models deconvolving the same input image, as well as two figures comparing both methods with their corresponding segmentations.

The segmentation and CARE models are part of an ongoing manuscript, so I'll post an appropriate DOI when we have a preprint/publication. Training data for both will be in the manuscript.

Thanks a lot!

![CARE_CAREAMICS_comp_subset](https://github.com/user-attachments/assets/66ecae98-8a5e-4c65-8e63-2616979b742a)
![CARE_CAREAMICS_comp](https://github.com/user-attachments/assets/07601a8f-c9e8-4bd1-b13d-f9aa92081e63)
![CARE_deconvolution](https://github.com/user-attachments/assets/d81549ed-cf63-4bae-a309-853f440e9ed6)
![CAREAMICS_deconvolution](https://github.com/user-attachments/assets/da623f74-34e5-4855-8a8e-10bb60670e3b)

**Comment by jdeschamps** [comment]
That looks pretty good indeed! Are those "tests" images that the network never saw during training?


We would be super interested in featuring it in the examples section of the website once your manuscript is out!

**Comment by OReyesMatte** [comment]
Thanks a lot! We'd be thrilled to have it as an example 😁 
So I realized that for the CAREAMICS model they are, but not for the CARE model. I'll make a new example figure. Is it okay if I keep posting them here? Or would you rather I send them to you via other way?

**Comment by jdeschamps** [comment]
Can you open a new issue with the "example" label? thanks!!

That way we would be able to find it easily again.

**Comment by jdeschamps** [comment]
Would require first to solve https://github.com/CAREamics/careamics/issues/264

**Comment by jdeschamps** [comment]
After internal discussions, we believe that it would require too much work to be able to load TF-based wieghts into the PyTorch models, especially when considering how long retraining from scratch would take for most applications.

Which was exemplified by your wonderful results!

So unless an important application comes along, we will not consider implementing that feature!
